### ISSUE TITLE
isoloate active jobs based on estimate and active flags, and possibly also if pins set = true

### ISSUE THREAD

**Issue by Cadmanick** [azithinkiam/brsgis_plugin#76]
Can you make a tool that allows us to see only jobs which are active, estimated or have pins not set? 

Can you add a new database table entry to be maintained as a new layer for this purpose, unless having some programmatic switching of layer isolation and styling is better suited.

I need to be able to visualize the location for scheduling.  to group jobs off peninsula, and local...  this could be a newly maintained layer that I can use to see this.  Also useful is possibly a visual styling based on job type and age of job.  So maybe border color/thickness blue/green to red for age, with red being older.  With jobtype being colored like the standard job type styles.  Maybe a hatching based on active, estimate and pinsets? 
1. can you do programmatic layer creation or linked/automated creation or does it need to be a report type of layer creation?
2. can you set the styling of this layer as requested?  seems valid to me but i go overboard with functionality usually...

:nP

**Comment by azithinkiam** [comment]
Is that logic good and does 53 sound like an accurate number for this criteria?

 

Definitely wouldn’t want to add a new table, but having a separate layer filtered as-needed should do the trick.  Styling will largely be up to you, but once it’s set it’s set…if you’re asking me to try to set up the styling, I don’t mind but it’ll probably include a good bit of back and forth.  You’ve spent way more time styling than I have, but I don’t think anything you’re suggesting is out of the question.  

 

I’m not following on the “layer creation” that you’re talking about in #1 though.  I’d think we’d just duplicate the jobs layer, then filter and style it the way you want.  Keep it off until you need to see it…then turn it on.

 

Short version:  I’m not seeing the need for a tool per se…just a new layer that shows you features filtered based on active/estimate/pins…with customized styling, obviously.

**Comment by azithinkiam** [comment]
try:
            if str(pins.currentText()) == '':

                if str(job_type.currentText()) == '':
                    pins.setCurrentText('N/A')
                elif str(job_type.currentText()) in ('BRS','BRSDP', 'SDP'):
                    pins.setCurrentText('NO')
            else:
                pass
        except Exception as e:
            pass
